stdlib/public/core/BridgeObjectiveC.swift

```diff
@@ -640,3 +641,200 @@ public func _getObjCTypeEncoding<T>(_ type: T.Type) -> UnsafePointer<Int8> {
 }
 
 #endif
+
+//===--- Bridging without the ObjC runtime --------------------------------===//
+
+#if !_runtime(_ObjC)
+
+/// Convert `x` from its Objective-C representation to its Swift
+/// representation.
+/// COMPILER_INTRINSIC
+@_inlineable // FIXME(sil-serialize-all)
+public func _forceBridgeFromObjectiveC_bridgeable<T:_ObjectiveCBridgeable> (
+  _ x: T._ObjectiveCType,
+  _: T.Type
+) -> T {
+  var result: T?
+  T._forceBridgeFromObjectiveC(x, result: &result)
+  return result!
+}
+
+/// Attempt to convert `x` from its Objective-C representation to its Swift
+/// representation.
+/// COMPILER_INTRINSIC
+@_inlineable // FIXME(sil-serialize-all)
+public func _conditionallyBridgeFromObjectiveC_bridgeable<T:_ObjectiveCBridgeable>(
+  _ x: T._ObjectiveCType,
+  _: T.Type
+) -> T? {
+  var result: T?
+  T._conditionallyBridgeFromObjectiveC (x, result: &result)
+  return result
+}
+
+public // SPI(Foundation)
+protocol _NSSwiftValue: class {
+  init(_ value: Any)
+  var value: Any { get }
+  static var null: AnyObject { get }
+}
+
+@usableFromInline
+internal class _SwiftValue {
+  @usableFromInline
+  let value: Any
```

Comment:
You could avoid double-indirecting using a small class hierarchy:

```
open class _SwiftValue {}

internal class _SwiftValueOf<T>: _SwiftValue {
  let value: T
  init(_ value: T) { self.value = value }
}

@_silgen_name("_swift_makeSwiftValue")
func _makeSwiftValue<T>(_ value: T) -> _SwiftValue { return _SwiftValueOf(value) }
```

This might also let you represent null as a singleton subclass instead of as a wrapper around a specific kind of `Optional`. Maybe you could also make it so that Foundation's preexisting `_SwiftValue` and `NSNull` are `Swift._SwiftValue` subclasses, so that they're substitutable for stdlib `_SwiftValue`s, instead of the `_NSSwiftValue`/`_SwiftValueFlavor` hacks too.

Reply:
We cannot reparent `NSNull`, and I would be loathe to reparent `_SwiftValue` for the reasons outlined below. Unless we reparent NSObject, which is possible but would be _really weird_.